### DISCUSSION TITLE
source folder

### DISCUSSION THREAD

**Discussion by ghost** [tighten/jigsaw#610] [Q&A] [answered]
this isn't any issues whatsoever, but I really like to know if there's a way jigsaw can maybe change the source folder to something else like
source -> public

**Comment by bakerkretzmar** [comment, marked as answer]
Yep, you can add a `build` key in your config and specify the `source` and `destination` directories there. E.g.:

```php
// config.php

return [
    'build' => [
        'source' => 'content',
        'destination' => 'public',
    ],
];
```

**Reply by ghost** [reply]
omg, finally, thank you! there is one thing i wanted to know if it actually works is the view system for jigsaw. normally i'm use to building sites with the following layout in the public folder which the build key works expected

is this possible, i know is a long shot.
```
public
   assets
     css
     js
   views
     about-me.blade.php
     index.blade.php
```
basically all php files are in the views folder.

**Reply by ghost** [reply]
so this is what i have, 
```
    'build' => [
        'source' => 'public/views',
    ],
 ```
 so apparently this works, but the assets will automatically go into that view folder, is there a key for the assets as well so that i can mvoe it one level up.

**Reply by bakerkretzmar** [reply]
No, the only keys are `source`, `destination`, and `views`. If you wanted to leave the source directory as-is, or make it the root `public` directory, you could put your Blade templates somewhere else using the `views` key instead.

**Reply by ghost** [reply]
how do i use the view key, can you provide me an example... appreciated.

**Reply by bakerkretzmar** [reply]
```php
// config.php

return [
    'build' => [
        // path to your content, like collections, etc.
        'source' => 'source',
        // path to your Blade views, like layouts, components, templates, etc.
        // defaults to the value of the `source` option above
        'views' => 'resources/views',
        // output path
        'destination' => 'dist',
    ],
];
```

**Reply by ghost** [reply]
awesome, thanks.

**Reply by ghost** [reply]
i di some test and to see if thhis actually works, but i'm having issues with the `views => 'resources/views`. it seems like is not picking up the blade templates. 

any idea why

**Reply by bakerkretzmar** [reply]
Can you give me some more details about your setup? Where is the Blade view you want to use and how are you 'registering' it for a collection (e.g. in the config, in the YAML frontmatter, etc)?

**Reply by ghost** [reply]
so to answer your question, this is what i have done.  This is how my config.php looks like right now
```
return [
   'build' => [ 
      'source' => 'public'
    ],
    'collections' => [
        'posts' => [
            'extends' => '_views.single',
            'section' => 'content',
            'author' => 'Benjamin Lu',
            'path' => 'archives/{date|Y/m/d}/{-title}',
            'sort' => '-date',
            'getDate' => function( $page ) {
                return Datetime::createFromFormat( 'U', $page->date );
            },
            'hasCategory' => function ( $page, $category ) {
                return collect( $page->categories )->contains( $category );
            },
        ],
        'categories' => [
            'extends' => '_views.category',
            'section' => 'content',
            'path' => '/category/{filename}',
            'name' => function( $page ) {
                return $page->getFilename();
            }
        ],
];
```
so my folder structure is inside of `public` folder is
```
public
  _views
    templates
      main.blade.php
    home.blade.php
    page.blade.php
  assets
  404.md - extends( '_views.page.blade.php' )
  about.md - extends( '_views.page.blade.php' )
  index.md - extends( '_views.home.blade.php' )
```
mainly I wanted to try to change the `public/_views` to `public/views` the public/views should be all `*.md`, `*.blade.php`, and `*.blade.md` The way I wanted is 
```
public
  assets
  views
    _posts
    _portfolio
   _categories
    templates
      main.blade.php
      home.blade.php
      page.blade.php
    404.md - extends( 'views.page.blade.php' )
    about.md - extends( 'views.page.blade.php' )
    index.md - extends( 'views.home.blade.php' )
```
i hope that make sense.

**Reply by bakerkretzmar** [reply]
Are you still having trouble with this?

I _think_ part of the issue is that your views folder is inside the source. If you keep it there, e.g. `public/_views`, I think you'll need to keep the `_` so that Jigsaw doesn't treat it like content. If you don't want the `_` in the name it'll have to go outside the source directory I think, something like this:

```
public
    assets
    _categories
        etc.
resources
    views
        main.blade.php
        etc.
```

**Reply by ghost** [reply]
nope, i got this working as expected by using _views now... so this is okay.

**Comment by ghost** [comment]
but i think right now. this is what i have on my repo
https://github.com/benlumia007/benlu.blog and the following structure in public is what i have right now. I just assume that anythign with `_` woudl be hidden anyways.
